src/viewProviders/search.ts

```diff
@@ -0,0 +1,74 @@
+import { FileDecoration, FileDecorationProvider, ThemeColor, Uri } from "vscode";
+import { ISearchable, isSearchable } from "../models/resource";
+
+/** Check if an item matches the provided search string. */
+export function itemMatchesSearch(item: ISearchable, searchStr: string): boolean {
```

Comment:
If the view being searched made sure to lowercase searchStr before the looped calls into `itemMatchesSearch()` and / or `traverseMatches()`, then whole milliseconds would be saved in not repeatedly lowercasing searchStr.

The truly squirrelbrained would make a branded type for lowercased strings, but that's a hairy tail too far, innit? Or isn't it?

Reply:
True, but would then force all callers to make sure it was `.toLowerCase()` which could make this more fragile